lib/beetle/queue_properties.rb

```diff
@@ -70,29 +67,24 @@ def set_queue_policy!(server, queue_name, options={})
 
       is_default_policy = definition == config.broker_default_policy
 
-      get_response = run_rabbit_http_request(request_url, get_request) do |http|
-        http.request(get_request, nil)
-      end
+      get_response = run_api_request(server, Net::HTTP::Get, request_path)
 
       case get_response.code
       when "200"
         response_body = JSON.parse(get_response.body) rescue {}
         same_policy = put_request_body.all? { |k,v| response_body[k] == v }
         if same_policy
           if is_default_policy
-            run_rabbit_http_request(request_url, delete_request) do |http|
-              http.request(get_request, nil)
```

Comment:
This issues a get_request but in the line above, the request was configured with a delete. 
I am not sure if that's a "bug" but I normalized it to just being a delete. 
This kind of error was invited by the previous design of the code, which was strangely round-about.
But again, I might be missing something crucial.